### PR TITLE
Add renovate config and fix errors

### PR DESCRIPTION
**Pull request by querti**
The new directory contains the renovate config and a kustomization file to create a configmap out of it. The git path to these resources is expected to be referenced in mintmaker infra-deployments definitions. Introducing renovate config changes can be done by changing the commit SHA in mintmaker infra-deployments.

I haven't split the config into presets. The main reason is that it uses another version control system, which means that we would need to maintain versioning for the global config as well as for all of its presets. This makes things needlesly complicated and doesn't provide any benefit. Secondly, the main reason for using presets is reusing the configuration by other users. I suggest to refrain from arbitrarily splitting the config file until we know which parts other people want to reuse.

The config itself was fixed to not contain any errors and warnings. List of changes:
- Remove outer quotes from prBodyDefinitions, as this field expects a JSON object. This is the only error in the config
- Change the branchName field to branchPrefix and branchTopic. The branchName field is deprecated.
- Change '"rebaseStalePrs": true' to '"rebaseWhen": "behind-base-branch"'. This change was suggested as a renovate migration.

Refers to CWFHEALTH-3210, CWFHEALTH-3206

### CODE REVIEW

File: renovate-config/kustomization.yaml
Comment (qixiang) [inline renovate-config/kustomization.yaml:6]:
```suggestion
  files:
  - renovate.json
  options:
    disableNameSuffixHash: true
```
Without disabling name suffix, the configmap will be created with name `renovate-config-c7mff5kd49` which has a hash suffix, MintMaker only looks for configmap `renovate-config`.

You check that by running:
```
$ kustomize build https://github.com/querti/mintmaker-config/renovate-config?ref=acf9a152cde1a40b96b652ba58cbd39726781919
```

File: renovate-config/renovate.json
Comment (qixiang) [inline renovate-config/renovate.json]: Could you also include the new manager has been added by https://github.com/redhat-appstudio/infra-deployments/pull/4035/files. Note: there is a json syntax issue will be fixed by https://github.com/redhat-appstudio/infra-deployments/pull/4096

**Review by qixiang — CHANGES_REQUESTED** [review]
I did a quick review, it generally looks good to me. I'll continue to review it more thoroughly tomorrow.

File: renovate-config/renovate.json
Comment (staticf0x) [inline renovate-config/renovate.json]: Also need to include this change: https://github.com/redhat-appstudio/infra-deployments/pull/4059

File: renovate-config/kustomization.yaml
Comment (querti) [reply renovate-config/kustomization.yaml:6]: Good point, added, thanks.

File: renovate-config/renovate.json
Comment (querti) [reply renovate-config/renovate.json]: Added along with fix.

File: renovate-config/renovate.json
Comment (querti) [reply renovate-config/renovate.json]: Added, thanks.

**Comment by querti** [comment]
Related PR in infra deployents repo: https://github.com/redhat-appstudio/infra-deployments/pull/4111
The CI will keep failing until this PR is merged though...

**Comment by qixiang** [comment]
> Related PR in infra deployents repo: [redhat-appstudio/infra-deployments#4111](https://github.com/redhat-appstudio/infra-deployments/pull/4111) The CI will keep failing until this PR is merged though...

Hi @querti , I realized we actually don't need to keep the config file in a dedicated repo, we can put the config file in `mintmaker` repo, in a directory under the `config/` dir, but don't include the kustomization of renovate in default kustomization, then we can manage them independently. For example:
```
  - https://github.com/konflux-ci/mintmaker/config/default?ref=9bb8d2a7dd571c2946fa3e5f011756bd35528661
  - https://github.com/konflux-ci/mintmaker/config/renovate?ref=...
```
then, the default manifests and the renovate config will be updated together for staging automatically, this is fine because we do want to apply latest changes to staging.
For production, since it's not automated, we can update them independently for default manifests and the renovate config.

Could you push this change to [mintmaker](https://github.com/konflux-ci/mintmaker/) repo, place the renovate config under `config/renovate`? Then we don't need to onboard the mintmaker-config repo to konflux to implement the automation of reference update (for staging). Sorry for being changing the solution on this.

**Comment by qixiang** [comment]
For the changes in PR, it looks good to me, I'll approve it when this is submitted to the mintmaker repo.

**Comment by staticf0x** [comment]
The config looks good, if we can keep it in one repo as @qixiang says (and delete this repo?) then even better.

**Comment by querti** [comment]
Closing, the config will reside in the mintmaker repo

**Comment by gnaponie** [comment]
@qixiang do you mind deleting this repo? Seems like it isn't needed. Thank you.

**Comment by qixiang** [comment]
> @qixiang do you mind deleting this repo? Seems like it isn't needed. Thank you.

@gnaponie  Yes, this repo can be deleted, but we don't have the permission, we probably need the help from `konflux-ci` org admins.

**Comment by gnaponie** [comment]
@arewm @gbenhaim @scoheb @hugares  could any of you help us deleting this repo? It isn't really needed anymore and we don't have permission to do so. Thank you.

**Pull request closed without merging**
